tikvrpc/tikvrpc.go

```diff
@@ -813,7 +813,10 @@ func SetContext(req *Request, region *metapb.Region, peer *metapb.Peer) error {
 		ctx.RegionEpoch = region.RegionEpoch
 	}
 	ctx.Peer = peer
-	if !AttachContext(req, ctx) {
+
+	// Shallow copy the context to avoid concurrent modification.
+	copyCtx := req.Context
+	if !AttachContext(req, &copyCtx) {
```

Comment:
How about to change `AttachContext(req *Request, ctx *kvrpcpb.Context)` to `AttachContext(req *Request, ctx kvrpcpb.Context)` to force shallow copy semantics?

Reply:
Great, done.